src/Calendar.jsx

```diff
@@ -77,12 +80,12 @@ var Calendar = React.createClass({
   propTypes,
 
   getInitialState: function(){
-    var value = this.inRangeValue(this.props.value);
+    var value = this.normalizeValue(this.props.value);
 
     return {
       selectedIndex: 0,
-      view:          this.props.initialView || 'month',
-      currentDate:   value ? new Date(value) : new Date()
+      view: this.props.initialView || 'month',
+      currentDateRange: value ? value : [new Date(), null]
```

Comment:
`currentDate` determines what month/year/decade to show in any of the views, so it should always just be a single date. even if the value is a range